### PR TITLE
#372 allow sortable(false) on fields in list view

### PR DESCRIPTION
**Pull request by RichardBradley**
Fixes https://github.com/marmelab/ng-admin/issues/372

(UPDATE: please ignore the rest of this comment, it refers to an old version of this PR. Jump straight to [the comment starting "OK, I have updated this pull request to include the necessary changes"](#issuecomment-205234237))

As discussed on that thread, this PR is much easier if I don't have to also modify `admin-config`.
However, it does look rather out of place next to all the fields with function getters & setters, so maybe a version which spans the two respositories would be better.

This version works for me though.

(Another benefit of the getter/setter approach is that it allows a "builder" style. This version is awkward to use:

```
articles.listView()
   .fields(
       nga.field('id'),
       function() {
            var headline = nga.field('headline');
            // Headline is an "analyzed" string field, so cannot be sorted in ES
            // https://www.elastic.co/guide/en/elasticsearch/guide/current/multi-fields.html
            headline.disableSort = true;
            return headline;
        }(),
        nga.field('publication_time', 'datetime'),
        nga.field('type')
    ])
```

whereas the getter/setter version would look like

```
articles.listView()
   .fields(
       nga.field('id'),
       nga.field('headline')
            // Headline is an "analyzed" string field, so cannot be sorted in ES
            // https://www.elastic.co/guide/en/elasticsearch/guide/current/multi-fields.html
            .disableSort(true),
        nga.field('publication_time', 'datetime'),
        nga.field('type')
    ])
```

)

### CODE REVIEW

**Comment by RichardBradley** [comment]
Taking a step back, why is `admin-config` in a separate git repos to `ng-admin`?
It would simplify the project a lot to merge them together; they are already very tightly coupled, so having them in separate repositories just adds administrative overhead.

**Comment by fzaninotto** [comment]
`admin-config` is also used by `react-admin`, that's why it's independent. 

Please update `admin-config` with a related PR to use a more usual function syntax.

**Comment by Fr4ncx** [comment]
** .disableSort(true),** Not work in my panel, why?

**Comment by RichardBradley** [comment]
> .disableSort(true), Not work in my panel, why?

Because this enhancement has not yet been implemented.
The issue is still marked "open" at the top -- subscribe to #372 for updates and try again when it is marked "closed".

**Comment by RichardBradley** [comment]
> Please update admin-config with a related PR to use a more usual function syntax.

Is there a guide on how best to do this?
I'll make the changes to `admin-config`; do I then need to bump its version and update the dependency reference in [`ng-admin/package.json`](https://github.com/marmelab/ng-admin/blob/master/package.json)? How will that work if the PR to `admin-config` has not been pushed yet?

**Comment by RichardBradley** [comment]
OK, I have updated this pull request to include the necessary changes to `admin-config`. Please could you take another look and merge if you are happy?

This now depends on https://github.com/marmelab/admin-config/pull/65 -- please merge and release that first, or this will fail. 

I have bumped the `admin-config` version and increased the version required by `ng-admin` accordingly.

I have not included changes to `build/`, as requested in the README

**Comment by RichardBradley** [comment]
(The CI build has failed because it cannot find the not-yet-published version of `admin-config` on which this change depends. This is inevitable with the current git repos structure, as briefly discussed above.)

**Comment by RichardBradley** [comment]
I have changed the syntax from the proposed "disabledSort(true)" to "`.sortable(false)`", as it better matches the other Field properties like "editable"

File: doc/reference/Field.md [hunk 1]
```diff
@@ -5,21 +5,21 @@ A field is the representation of a property of an entity.
 * [General Field Settings](#general-field-settings)
```
Comment (fzaninotto) [inline doc/reference/Field.md]: please revert these changes, they are not related to this PR, and break the ng-admin book on GitBook (i.e. the official doc)

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "url": "git://github.com/marmelab/ng-admin.git"
   },
   "devDependencies": {
-    "admin-config": "^0.11.0",
+    "admin-config": "^0.11.1",
```
Comment (fzaninotto) [inline package.json]: I have released 0.12.0

As a side note, 0.0.x is for bugfixes, 0.x is for new features. `.sortable()` is a new feature. That's called semver.

File: doc/reference/Field.md (hunk 1 above)
Comment (RichardBradley) [inline doc/reference/Field.md]: OK, will do.

The links are currently broken on GitHub, see e.g. https://github.com/marmelab/ng-admin/blob/master/doc/reference/Field.md#-date-field-type

They do work on the "GitBook" version, you're right: http://ng-admin-book.marmelab.com/doc/reference/Field.html#-date-field-type

It's a shame that the two differ. I hadn't spotted the "GitBook" version; I'd been referring to the documentation on GItHub directly up until now. If the GitBook version is the official one, it should win, of course.

File: package.json (hunk 2 above)
Comment (RichardBradley) [inline package.json]: Thanks, I'll update.

**Comment by RichardBradley** [comment]
I have pushed a new version with the requested changes (and rebased to account for other changes in the meantime).

**Comment by fzaninotto** [comment]
This PR is OK, just needs rebasing

**Comment by fzaninotto** [comment]
Superseded by #1260
